src/lib/common/include/sol-platform.h

```diff
@@ -156,6 +160,75 @@ int sol_platform_restart_service(const char *service);
 
 int sol_platform_set_target(const char *target);
 
+struct sol_mount_point {
+    char *dev;
+    char *mpoint;
+    char *fstype;
+    unsigned long flags;
+    char *opts;
+    bool noauto;
+};
+
+static inline void
+sol_mount_point_free(struct sol_mount_point *mp)
+{
+    if (!mp)
+        return;
+
+    free(mp->dev);
+    free(mp->mpoint);
+    free(mp->fstype);
+    free(mp->opts);
+    free(mp);
+}
+
+/**
+ * List mounted mount points, if @c hplug_only is provided this function will list
+ * only mount points mounted by us, all mounted mount points are listed otherwise.
+ *
+ * @param hplug_only List only mount points only mounted by us?
```

Comment:
what's the point to list other mount points? Here we care about non-system mount points, so only hot plug (auto mounted). Those that are system-specific, such as linux-micro setup (/proc, /sys, /dev...) shouldn't show here. I'd consider /etc/fstab the same.